[BUGFIX] Hive temporary tables creation fix

jaume-ferrarons opened the pull request:
This PR contains minor fixes in order to get Great Expectations running in Hive.

Changes proposed in this pull request:
- Do not quote Hive table names in temporary table creation. Hive temporary table creation used the default entry quoting table names, but that's not supported by Apache Hive. 
- ~~Reduce redundant code: `self.sql_engine_dialect.name.lower()` was executed several times. The result is computed once and stored in a variable.~~
- ~~Make `DataConnectorConfig` and `DatasourceConfig` serializable. Required to be able to store Hive's data connection objects. I had to copy `to_json_dict`'s implementation, as there are circular dependencies. Fixing that, I think, should be done in a different PR.~~

### Definition of Done


- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Commit:
fix create temporary table for hive

Commit:
object serialization

jaume-ferrarons committed:
Merge remote-tracking branch 'upstream/develop' into fix/hive_connection

jaume-ferrarons committed:
Add removed new line

jaume-ferrarons committed:
fix lint

jaume-ferrarons commented:
@cla-bot check

talagluck commented:
Thanks so much for the contribution, @jaume-ferrarons! We will review and be in touch.

Commit:
Merge branch 'develop' into fix/hive_testing_tmp_tables_and_persistance

Commit:
Fix test

rdodev commented:
@cdkini can you please take a look at this PR?

alexsherstinsky commented on great_expectations/execution_engine/sqlalchemy_batch_data.py:
```diff
@@ -217,11 +224,10 @@ def _create_temporary_table(
+        elif dialect == GESqlDialect.MYSQL or dialect == GESqlDialect.HIVE:
             stmt = f"CREATE TEMPORARY TABLE {temp_table_name} AS {query}"
         elif dialect == GESqlDialect.MSSQL:
+
```
Very minor: If we do not need this blank line, let us not add it.  Thank you.

jaume-ferrarons replied:
Removed 👍

Commit:
Remove black line

cdkini commented:
@jaume-ferrarons would you happen to have some more information (or a test case) around why you updated the config class to inherit from `SerializableDictDot`? Hoping to get a better understanding of that so we can prevent future issues.

Commit:
Revert serialization

Commit:
GESqlDialect handle byte strings

jaume-ferrarons commented:
@cdkini The issues I observed occurred when storing the results of an expectation generated with the profiling when running `great_expectations suite new`. I tried to reproduce the error with the current status of `develop` but I couldn't, I guess it got fixed somehow. So, I've just reverted the changes.

Though, I've found that `GESqlDialect` is instantiated in different places, so I've moved the logic of handling the byte strings to the class and added a simple test.

Commit:
Quote Hive temporary table name

jaume-ferrarons commented:
I was pointed out that it's better to use back quotes for table names in Hive. I've introduced that change.

I was also pointed out that there's [this SQLAlchemy method](https://docs.sqlalchemy.org/en/14/core/internals.html?highlight=dialect#sqlalchemy.sql.compiler.IdentifierPreparer) to handle the quoting. It may be useful to switch to it in the future.

donaldheppner committed:
Merge branch 'develop' into fix/hive_testing_tmp_tables_and_persistance

cdkini commented:
@jaume-ferrarons thanks for the explanation! I'm approving the PR 😄 

I'll add a quick type annotation to pass out linter checks and set it on auto-merge so we can get your changes in. Thanks so much for all the help here!

Commit:
chore: update type hint
